### DISCUSSION TITLE
windows如何做成后台运行

### DISCUSSION THREAD

**Discussion by 946106004** [messense/aliyundrive-webdav#530] [Q&A] [answered]
### 问题描述

我使用nssm把aliyundrive-webdav.exe做成windows service运行，但是进程可以看到，端口却没有监听，也无法链接。

### 重现步骤

1.nssm把aliyundrive-webdav.exe做成了service服务，并设置开机启动
2.重启机器，发现进程存在，但是netstat看不到端口，raildrive链接不上。
3.改为手动双击启动，cmd窗口出来，只要不关闭，raildrive可以正常连接。

### 版本

最新版本

### 运行平台

win11

### refresh token 获取方式

二维码

### 日志

没有日志

**Comment by messense** [comment]
不用 Windows，自行排查吧。

**Comment by ixuzone** [comment, marked as answer]
提供一种方案
1.将启动命令编写为 aliyundrive-webdav.bat 脚本
2.编写 vbs 脚本后台启动
脚本内容为
set ws=WScript.CreateObject("WScript.Shell") 
ws.Run "目标脚本  /start",0 
示例：
set ws=WScript.CreateObject("WScript.Shell") 
ws.Run "D:\Script\aliyundrive-webdav.bat  /start",0 
3.将 vbs 脚本放到 C:\ProgramData\Microsoft\Windows\Start Menu\Programs\Startup 实现开机自启
